test/spec/modules/sovrnAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,174 @@
+    beforeEach(() => {
+      adaptermanager.enableAnalytics({
+        provider: 'sovrn',
+        options: {
+          affiliateId: 123
+        }
+      });
+      sinon.spy(sovrnAnalyticsAdapter, 'track');
+    });
+    afterEach(() => {
+      sovrnAnalyticsAdapter.disableAnalytics();
+      sovrnAnalyticsAdapter.track.restore();
+    });
+    let auctionStartTimestamp = Date.now();
+    let auctionId = '123-123-123-123';
+    let timeout = 3000;
+    let auctionInit = {
+      timestamp: auctionStartTimestamp,
+      auctionId: auctionId,
+      timeout: timeout
+    };
+    let bidderCode = 'sovrn';
+    let bidderRequestId = '123bri';
+    let adUnitCode = 'div';
+    let bidId = 'bidid';
+    let bidRequested = {
+      auctionId: auctionId,
+      auctionStart: auctionStartTimestamp,
+      bidderCode: bidderCode,
+      bidderRequestId: bidderRequestId,
+      bids: [
+        {
+          adUnitCode: adUnitCode,
+          auctionId: auctionId,
+          bidId: bidId,
+          bidder: bidderCode,
+          bidderRequestId: '10340af0c7dc72',
+          sizes: [[300, 250]],
+          startTime: auctionStartTimestamp + 100,
+          transactionId: '7aafa3ee-a80a-46d7-a4a0-cbcba463d97a'
+        }
+      ],
+      doneCbCallCount: 1,
+      start: auctionStartTimestamp,
+      timeout: timeout
+    };
+    it('should create auctiondata record from init ', function () {
+      events.emit(constants.EVENTS.AUCTION_INIT, auctionInit);
+
+      let auctionData = sovrnAnalyticsAdapter.getAuctions();
+      let currentAuction = auctionData['123-123-123-123'];
+      assert(currentAuction);
+      assert.equal(currentAuction.auction.auctionId, '123-123-123-123');
+      assert.equal(currentAuction.auction.aid, 123);
+    });
+    it('should create a bidrequest object ', function() {
+      events.emit(constants.EVENTS.AUCTION_INIT, auctionInit);
+      events.emit(constants.EVENTS.BID_REQUESTED, bidRequested);
+
+      let auctionData = sovrnAnalyticsAdapter.getAuctions();
+      let currentAuction = auctionData['123-123-123-123'];
+      assert(currentAuction);
+      let requests = currentAuction.auction.requests;
+      assert(requests);
+      assert.equal(requests.length, 1);
+      assert.equal(requests[0].bidderCode, bidderCode);
+      assert.equal(requests[0].bidderRequestId, bidderRequestId);
+      assert.equal(requests[0].timeout, timeout);
+      let bids = requests[0].bids;
+      assert(bids);
+      assert.equal(bids.length, 1);
+      assert.equal(bids[0].bidId, bidId);
+      assert.equal(bids[0].bidder, bidderCode);
+      assert.equal(bids[0].transactionId, '7aafa3ee-a80a-46d7-a4a0-cbcba463d97a');
+      assert.equal(bids[0].sizes.length, 1);
+      assert.equal(bids[0].sizes[0][0], 300);
+      assert.equal(bids[0].sizes[0][1], 250);
+      expect(requests[0]).to.not.have.property('doneCbCallCount');
+      expect(requests[0]).to.not.have.property('auctionId');
+    })
+  })
+});
```

Comment:
Is it possible to add some tests around Bid Adjustments and bids that don't match (whatever would end up in the `unsynced` array)?

Reply:
Yep. Added both.